keel/leads/serializers.py

```diff
@@ -0,0 +1,17 @@
+from rest_framework import serializers
+from rest_framework import validators
+from .models import CustomerLead
+
+
+class CustomerLeadSerializer(serializers.ModelSerializer):
+
+    email = serializers.EmailField(required=True, 
+                                    validators=[validators.UniqueValidator(queryset=CustomerLead.objects.all())])
```

Comment:
objects.all is a very heavy query for a single request, instead always use a filter query.
btw we dont need a unique filter on this.
we could have multiple leads , EG: say a lead drops idea to immigrate and we make it inactive...after 2 years he decides to immigrate again, we will use a fresh lead then for the same email

Reply:
this would mean I would have to remove the unique parameter in the CustomLead model email field, right?

Reply:
yes

Reply:
updated